relays/ethereum/src/ethereum_client.rs

```diff
@@ -439,30 +453,54 @@ struct EthereumHeadersSubmitter {
 
 #[async_trait]
 impl HeadersSubmitter for EthereumHeadersSubmitter {
-	async fn is_header_incomplete(&self, header: &QueuedSubstrateHeader) -> Result<bool> {
-		let (encoded_call, call_decoder) =
-			bridge_contract::functions::is_incomplete_header::call(header.header().encode());
+	async fn is_headers_incomplete(
+		&self,
+		header1: &QueuedSubstrateHeader,
+		header2: Option<&QueuedSubstrateHeader>,
+		header3: Option<&QueuedSubstrateHeader>,
+		header4: Option<&QueuedSubstrateHeader>,
+	) -> Result<usize> {
+		let (encoded_call, call_decoder) = bridge_contract::functions::is_incomplete_headers::call(
+			header1.header().encode(),
+			header2.map(|header2| header2.header().encode()).unwrap_or_default(),
+			header3.map(|header3| header3.header().encode()).unwrap_or_default(),
+			header4.map(|header4| header4.header().encode()).unwrap_or_default(),
+		);
 		let call_request = CallRequest {
 			to: Some(self.contract_address),
 			data: Some(encoded_call.into()),
 			..Default::default()
 		};
 
 		let call_result = self.client.eth_call(call_request).await?;
-		let is_incomplete = call_decoder.decode(&call_result.0)?;
+		let incomplete_index: U256 = call_decoder.decode(&call_result.0)?;
+		if incomplete_index > 4.into() {
```

Comment:
```suggestion
		if incomplete_index > MAX_SUPPORTED_HEADERS.into() {
```
Please move to a constant.